src/components/InterfaceNotification.tsx

```diff
@@ -1,27 +1,114 @@
 import React from "react";
-import { Spinner } from "react-bootstrap";
+import { OverlayTrigger, Spinner, Tooltip } from "react-bootstrap";
 import { AppSettings } from "../config/Variables";
 import { Locale } from "../config/Locale";
+import { Environment } from "../config/Environment";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import green from "@mui/material/colors/green";
+
+const theme = createTheme({
+  palette: {
+    success: { main: green[600] },
+  },
+});
 
 interface Props {
-  active: boolean;
+  loading: boolean;
   message: string;
   error: boolean;
   retry: boolean;
   performTransaction: (...queries: string[]) => void;
+  handleStatus: Function;
+  status: string;
+}
+
+interface State {
+  connection: boolean;
 }
 
-export default class InterfaceNotification extends React.Component<Props> {
+export default class InterfaceNotification extends React.Component<
+  Props,
+  State
+> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      connection: true,
+    };
+    this.setStatus = this.setStatus.bind(this);
+  }
+
+  componentDidMount() {
```

Comment:
One reason I will create new class components is if I need to use joints, the diagram rendering component - for some reason, it just seems to work better with initialization using componentDidMount instead of useEffect. In this case, it's just an old component that I didn't bother rewriting to an FC. But yes - I generally try to write FCs as much as possible now.